gbpservice/neutron/services/servicechain/drivers/oneconvergence_intercloud_driver.py

```diff
@@ -213,25 +262,56 @@ def _fetch_template_and_params(self, context, sc_instance,
 
                 config_param_values['RouterId'] = routers[0]['id']
 
+                provider_subnet = self._core_plugin.get_subnet(context._plugin_context, provider_subnet_id)
+
                 desc = 'fip=' + floating_ip + ";" + "tunnel_local_cidr=" + provider_subnet['cidr']
                 stack_params['ServiceDescription'] = desc
+                LOG.debug(_("Creation insert_type %s, sc_instance: %s, ptgs: %s)"), insert_type, sc_instance, ptgs)
 
         else:
-            provider_pt = self.create_pt(context, provider_ptg_id)
-            (service_instance_id, hotplugged_port_list,
-             stitching_port_ip) = (
-                        self.svc_mgr.create_service_instance(
-                            context._plugin_context,
-                            instance_type,
-                            svc_mgmt_pt["port_id"],
-                            self._get_ptg_subnet(context, provider_ptg_id),
-                            #cons_subnet_id=consumer_ptg_subnet_id,
-                            cons_subnet_id=None,
-                            port_id=provider_pt["port_id"]))
+            LOG.debug(_("Creation insert_type %s, sc_instance: %s, ptgs: %s)"), insert_type, sc_instance, ptgs)
+            subnet = self._core_plugin.get_subnet(context._plugin_context, provider_ptg_subnet_id)
+            network_id = subnet['network_id']
+            attrs = {'port': {'tenant_id': admin_context.tenant_id,
+                     'name': 'LB_provider_port',
+                     'network_id': network_id,
+                     'fixed_ips': [{"subnet_id": provider_ptg_subnet_id}],
+                     'port_security_enabled': True,
+                     'admin_state_up': True,
+                     #'security_groups': [{"id": sg_grp_id}],
+                     'mac_address': attributes.ATTR_NOT_SPECIFIED,
+                     'device_id': '',
+                     'device_owner': ''
+                    }}
+            with admin_context.session.begin(subtransactions=True):
+                lb_prov_port = self._core_plugin.create_port(
+                    admin_context, attrs)
+            #provider_pt = self.create_pt(context, provider_ptg_id)
+            service_instance_id = self.svc_mgr.create_service_instance(
+                            context=context._plugin_context,
+                            tenant_id=context._plugin_context.tenant_id,
+                            service_chain_instance_id=sc_instance_id,
+                            service_type=instance_type,
+                            provider_network_port=lb_prov_port['id'],
+                            consumer_network_port=consumer_port_id,
+                            stitching_network_port=stitching_port_id,
+                            management_port=svc_mgmt_pt["port_id"],
+                            mgmt_net_id=svc_mgmt_network_id,
+                            insert_type=insert_type)
+
+            floating_ip = self.svc_mgr.get_service_floating_ip(
```

Comment:
General approach if for fip to be retrieved from plugin., not passed via desc. Suggest to move fip call to plugin.